src/app/core/store/projects/projects.effects.ts

```diff
@@ -143,10 +146,26 @@ export class ProjectsEffects {
     withLatestFrom(this.store$.select(selectProjectsCodes)),
     flatMap(([{project, oldProject}, projectCodes]) => {
       const actions: Action[] = [new ProjectsAction.UpdateSuccess({project: {...project, id: project.id}})];
-      if (projectCodes) {
-        const codes = projectCodes[project.organizationId].map(code => code === oldProject.code ? project.code : code);
+      const codesByOrg = projectCodes && projectCodes[project.organizationId];
+      if (codesByOrg) {
+        const codes = codesByOrg.map(code => code === oldProject.code ? project.code : code);
         actions.push(new ProjectsAction.GetCodesSuccess({organizationId: project.organizationId, projectCodes: codes}));
       }
+
+      const paramMap = RouteFinder.getFirstChildRouteWithParams(this.router.routerState.root.snapshot).paramMap;
+      const projCodeInRoute = paramMap.get('projectCode');
+
+      if (projCodeInRoute && projCodeInRoute === oldProject.code && project.code !== oldProject.code) {
+        const paths = this.router.routerState.snapshot.url.split('/').filter(path => path);
+        const index = paths.indexOf(oldProject.code, 3);
+        if (index !== -1) {
+          paths[index] = project.code;
+          actions.push(new RouterAction.Go({path: paths}));
+          // TODO extract as
```

Comment:
As what? :)

Reply:
This TODO comment is also in `organizations.effects.ts` and I don't know what it means. So I decided to leave it here too.